lib/command/src/Obelisk/Command/Deploy.hs

```diff
@@ -150,13 +159,88 @@ deployPush deployPath getNixBuilders = do
+      putLog Notice $ "Creating keystore: " <> T.pack keystorePath
+      putLog Notice "Enter a keystore password: "
+      keyStorePassword <- liftIO $ withEcho False getLine
+      putLog Notice "Re-enter the keystore password: "
+      keyStorePassword' <- liftIO $ withEcho False getLine
+      unless (keyStorePassword' == keyStorePassword) $ failWith "passwords do not match"
+      let keyToolConf = KeytoolConfig
+            { _keytoolConfig_keystore = keystorePath
+            , _keytoolConfig_alias = "obelisk"
+            , _keytoolConfig_storepass = keyStorePassword
+            , _keytoolConfig_keypass = keyStorePassword
+            , _keytoolConfig_dname = "CN=example.com, OU=engineering, O=obelisk, L=LinuxLand, S=NY, C=US" -- TODO Read these from config?
+            }
+      createKeystore root $ keyToolConf
+      liftIO $ BSL.writeFile keytoolConfPath $ encode keyToolConf
+    checkKeytoolConfExist <- liftIO $ doesFileExist keytoolConfPath
+    unless checkKeytoolConfExist $ failWith "Missing android KeytoolConfig"
+    keytoolConfContents <- liftIO $ BSL.readFile keytoolConfPath
+    liftIO $ putStrLn $ show keytoolConfContents
+    releaseKey <- case eitherDecode keytoolConfContents of
+      Left err -> failWith $ T.pack err
+      Right conf -> return $ renderAttrset $ keytoolToAndroidConfig conf
+    result <- nixCmd $ NixCmd_Build $ def
+      & nixBuildConfig_outLink .~ OutLink_None
+      & nixCmdConfig_target .~ Target
+        { _target_path = Nothing
+        , _target_attr = Nothing
+        , _target_expr = Just $ "with (import " <> srcDir <> " {}); "  <> platform <> ".frontend.override (drv: { releaseKey = (if builtins.isNull drv.releaseKey then {} else drv.releaseKey) // " <> T.unpack releaseKey <> "; })"
+        }
+    putLog Notice $ T.pack $ "Your recently built android apk can be found at the following path: " <> (show result)
+    callProcessAndLogOutput (Notice, Error) $ proc (result </> "bin" </> "deploy") mobileArgs
+  where
+    withEcho showEcho f = do
+      prevEcho <- hGetEcho stdin
+      bracket_ (hSetEcho stdin showEcho) (hSetEcho stdin prevEcho) f
+
+data KeytoolConfig = KeytoolConfig
+  { _keytoolConfig_keystore :: FilePath
+  , _keytoolConfig_alias :: String
+  , _keytoolConfig_storepass :: String
+  , _keytoolConfig_keypass :: String
+  , _keytoolConfig_dname :: String
+  } deriving (Show, Generic)
+
+instance FromJSON KeytoolConfig
+instance ToJSON KeytoolConfig
+
+createKeystore :: MonadObelisk m => FilePath -> KeytoolConfig -> m ()
+createKeystore root config = do
+  let expr = "with (import " <> toImplDir root <> ").reflex-platform.nixpkgs; pkgs.mkShell { buildInputs = [ pkgs.jdk ]; }"
+  callProcessAndLogOutput (Notice,Notice) $ (proc "nix-shell" ["-E" , expr, "--run" , keytoolCmd]) { cwd = Just root }
+  where
+    keytoolCmd = unwords $ "keytool" :
+      [ "-genkeypair -noprompt"
+      , "-keystore", _keytoolConfig_keystore config
+      , "-keyalg RSA -keysize 2048 -validity 10000"
```

Comment:
Where is this `10000` coming from?

Reply:
I believe this value came from the android example https://developer.android.com/studio/publish/app-signing#signing-manually

Reply:
OK, yeah, `If you plan to support upgrades for an app, ensure that your app signing key has a validity period that exceeds the expected lifespan of that app. A validity period of 25 years or more is recommended. When your key's validity period expires, users will no longer be able to seamlessly upgrade to new versions of your app. `.  See also https://stackoverflow.com/questions/14272794/how-to-create-an-android-keystore-rsa-key-with-infinite-validity

I think we should go with 1 million days.